### PR TITLE
[KOGITO-196] Introducing kogito-data-index image

### PR DESCRIPTION
**Pull request by ricardozanini**
See:
https://issues.jboss.org/browse/KOGITO-196

@spolti Although we have another related [PR](https://github.com/kiegroup/kogito-runtimes/pull/135) and some more tests to run within a cluster, I'd like to hear from you about this change, mainly with the new module structure.

Thanks!

### CODE REVIEW

File: s2i/README.md [hunk 1]
```diff
@@ -49,6 +49,10 @@ Image location: quay.io/kiegroup/kogito-springboot-ubi8:latest
 
 For more details have a look at [README.md](kogito-springboot-ubi8/README.md)
 
+## Data Index Service
+
+Runtime image that is responsibile for just running the [Data Index Service](https://github.com/kiegroup/kogito-runtimes/wiki/Data-Index-Service). Sonely used by the [Kogito Operator](https://github.com/kiegroup/kogito-cloud-operator) to deploy data index capabilities to Kogito Services.
```
Comment (sutaakar) [inline s2i/README.md]:
```suggestion
Runtime image that is responsibile for just running the [Data Index Service](https://github.com/kiegroup/kogito-runtimes/wiki/Data-Index-Service). Solely used by the [Kogito Operator](https://github.com/kiegroup/kogito-cloud-operator) to deploy data index capabilities to Kogito Services.
```

**Comment by ricardozanini** [comment]
@mswiderski 
Here's the steps to work with the Data Index Image locally:

1. Pull the image from my repo: `docker pull quay.io/ricardozanini/kogito-data-index:latest`
2. Run Spotify Kafka locally: `docker run -p 2181:2181 -p 9092:9092 --env ADVERTISED_HOST=<host ip> --env ADVERTISED_PORT=9092 spotify/kafka`
2.1 To get your host ip, you can run the image with "localhost" in the ip and then run `docker exec $(docker ps -q | head -1) /sbin/ip route|awk '/default/ { print $3 }';`
3. Spin up a Infinispan server (10.0.0-CR1): `./server.sh -b 0.0.0.0`
4. Run the data index image: `docker run --rm -it -p 8180:8180  -e mp_messaging_incoming_kogito_processinstances_events_bootstrap_servers=<hostip>:9092 -e QUARKUS_INFINISPAN_CLIENT_SERVER_LIST=<hostip>:11222 quay.io/kiegroup/kogito-data-index`

Done!

Please let me know if you have any issues.

File: s2i/kogito-data-index-overrides.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+packages:
+  install:
+  - java-1.8.0-openjdk
+
+run:
+  user: 1001
+  cmd: 
+    - "/home/kogito/bin/start"
+  workdir: "/home/kogito/bin"
```
Comment (spolti) [inline s2i/kogito-data-index-overrides.yaml]: workdir seems unnecessary here, or will this image need it, copy artifacts, etc?

File: s2i/modules/kogito-data-index/module.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+schema_version: 1
+name: kogito-data-index
+version: "0.3.0"
+
+envs:
```
Comment (spolti) [inline s2i/modules/kogito-data-index/module.yaml]: we might want to move this env to the main image.yaml descriptor?

**Review by spolti — CHANGES_REQUESTED** [review]

File: s2i/modules/kogito-data-index/module.yaml (hunk 3 above)
Comment (ricardozanini) [reply s2i/modules/kogito-data-index/module.yaml]: I removed since we won't need it anymore

File: s2i/kogito-data-index-overrides.yaml (hunk 2 above)
Comment (ricardozanini) [reply s2i/kogito-data-index-overrides.yaml]: agreed!

**Review by spolti — APPROVED** [review]

**Comment by mswiderski** [comment]
just tested that with kogito travel agency app and it works fine

```
docker run --rm -it 
-p 8180:8180 
-v /temp/kogito:/tmp/protofiles 
-e KOGITO_PROTOBUF_WATCH=true 
-e KOGITO_PROTOBUF_FOLDER=/tmp/protofiles 
-e MP_MESSAGING_INCOMING_KOGITO_PROCESSINSTANCES_EVENTS_BOOTSTRAP_SERVERS=HOST:9092 
-e QUARKUS_INFINISPAN_CLIENT_SERVER_LIST=HOST:11222 
quay.io/ricardozanini/kogito-data-index:latest
```

so it does look good to me and ready to be merged

**Review by mswiderski — APPROVED** [review]

**Pull request merged**